src/index.js

```diff
@@ -0,0 +1,210 @@
+  toDoInput.classList.add('to-do-input');
+  toDoInput.dataset.complete_list = listItem.index;
+  mainElement.appendChild(toDoInput);
+
+  // Handling editing task event
+  toDoInput.addEventListener('click', (e) => {
+    toDoInput.removeAttribute('readonly');
+    const IndexToEdit = e.target.dataset.complete_list;
+    toDoInput.addEventListener('change', () => {
+      const updatedTask = toDoInput.value;
+      EditTask(IndexToEdit, updatedTask);
+      toDoInput.setAttribute('readonly', true);
+    });
+  });
+
+  // Creating a delete icon
+
+  const deleteBtn = document.createElement('i');
+  deleteBtn.classList.add('fa-solid');
+  deleteBtn.id = listItem.index;
+  deleteBtn.classList.add('fa-ellipsis-vertical');
+
+  // Handling delete Event
+
+  deleteBtn.addEventListener('click', (e) => {
+    e.preventDefault();
+    if (deleteBtn.classList.contains('fa-trash-can')) {
+      deleteBtn.parentElement.remove();
+      removeFromStorage(e.target.id);
+    } else {
+      deleteBtn.classList.remove('fa-ellipsis-vertical');
+      deleteBtn.classList.add('fa-trash-can');
+    }
+  });
+
+  mainElement.appendChild(deleteBtn);
+
+  return mainElement;
+};
+
+// Appending the generated list to the html file
+
+const appendItems = () => {
+  myListItems.innerHTML = '';
+  const storage = getBookStorage();
+  storage.forEach((item) => {
+    const toDoItem = AddToDoListItems(item);
+    myListItems.appendChild(toDoItem);
+  });
+};
+
+// clear input
+
+const clearInputs = () => {
+  toDoItem.value = '';
+};
+
+// Adding tasks to local storage
+
+const addToStorage = (item) => {
+  const storage = getBookStorage();
+  storage.push(item);
+  localStorage.setItem('taskList', JSON.stringify(storage));
+  clearInputs();
+};
+
+// A function to add a new task
+
+const addTask = (task, complete = false, index) => {
+  const toDoTask = task;
+  const toDoComplete = complete;
+  const toDoIndex = index;
+
+  const newTask = {
+    completed: toDoComplete,
+    index: toDoIndex,
+    description: toDoTask,
+  };
+
+  addToStorage(newTask);
+  appendItems();
+};
+
+// Adding a new task from the input in the index.html file
+
+const addEvent = () => {
+  const taskStorage = getBookStorage();
+  if (!toDoItem.value) {
+    return;
+  }
+  addTask(toDoItem.value, false, taskStorage.length);
+};
+
+toDoItem.addEventListener('keydown', (e) => {
+  if (e.key === 'Enter') {
+    addEvent();
+  }
+});
+
+// Handling clear all event
+
+const clearAll = () => {
+  const storage = getBookStorage();
+  const newTaskArray = storage.filter((task) => task.completed === false);
+  localStorage.setItem('taskList', JSON.stringify(newTaskArray));
+};
+
+clearAllBtn.addEventListener('click', () => {
+  clearAll();
+  appendItems();
+});
+
+const refreshStatus = () => {
+  const storage = getBookStorage();
+  storage.forEach((stored) => {
+    stored.completed = false;
+  });
+  localStorage.setItem('taskList', JSON.stringify(storage));
+};
```

Comment:
- Great Job implementing the `clear all completed` functionality. 💪🏽 

- I'd like to kindly suggest that after clearing all completed items and refreshing the page. The index for the rest of the todos needs to be updated. Can we fix that?

This is what I mean 👇🏽 

| Local storage on your todoList |
| :---:   | 
|<img width="1399" alt="Screenshot 2023-01-21 at 09 26 28" src="https://user-images.githubusercontent.com/99246359/213874141-a8ab9d48-d5bd-46a6-b168-56058ba3fbbb.png">  |